app/models/upload.rb

```diff
@@ -105,6 +107,12 @@ def self.default_options(csv_type)
     Rails.application.config.csv_defaults[csv_type] || Rails.application.config.csv_defaults['generic']
   end
 
+  def self.fetching_for?(csv_type)
+    Upload.where(ok: false, completed_at: nil, csv_type: csv_type)
+          .order(updated_at: :desc)
```

Comment:
You're just checking that a record exists.  `order` doesn't matter.

Reply:
removed